common/stats/stats.go

```diff
@@ -0,0 +1,479 @@
+// This package borrows from go-metrics so we can avoid exposing dependencies,
+// and to give callers more flexibility in recording and formatting scoped stats.
+//
+// Original license: github.com/rcrowley/go-metrics/blob/master/LICENSE
+//
+package stats
+
+import (
+	"encoding/json"
+	"log"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/rcrowley/go-metrics"
+)
+
+// For testing.
+var Time StatsTime = DefaultStatsTime()
+
+// Stats users can either reference this global receiver or construct their own.
+var CurrentStatsReceiver StatsReceiver = NilStatsReceiver()
+
+// Overridable instrument creation.
+var NewCounter func() Counter = newMetricCounter
+var NewGauge func() Gauge = newMetricGauge
+var NewGaugeFloat func() GaugeFloat = newMetricGaugeFloat
```

Comment:
You two are way ahead of me in this conversation. I'm reading it but have frightfully little to add (other than "can you explain this to someone who hasn't thought about it much?")